### PR TITLE
Design document for validation

### PR DESCRIPTION
**Pull request by cheld**
I have created a short design doc for validation of user input. The document collects the discussion from the corresponding issues and PRs.

The synchronous validation is already implemented in the create namespace dialog, the async validation is not implemented anywhere, yet.

### CODE REVIEW

File: docs/design/validation.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Validation
+
+## Problem
+Kubernetes has restrictions for the possible values of resources. Therefore, Dashboard must
+validate user input.
+
+Please note: some input fields of Dashboard are used to create multiple resources in Kubernetes.
```
Comment (cheld) [inline docs/design/validation.md]: Basically, this is correct. However, I have not investigated if the value restrictions are aligned or not. But, in any case one should be aware of potential problems.

**Comment by cheld** [comment]
@zreigz can you have a look

**Comment by zreigz** [comment]
yes, sure

**Comment by zreigz** [comment]
LGTM

**Pull request closed without merging**